src/vis/vis.js

```diff
@@ -262,34 +259,20 @@ var VisModel = Backbone.Model.extend({
     this.settings.set(settings);
   },
 
-  _onMapInstanceCreated: function () {
-    this._analysisPoller.reset();
-    this._analysisCollection.each(function (analysisModel) {
-      analysisModel.unbind('change:status', this._onAnalysisStatusChanged, this);
-      if (analysisModel.url() && !analysisModel.isDone()) {
-        this._analysisPoller.poll(analysisModel);
-        this.trackLoadingObject(analysisModel);
-        analysisModel.bind('change:status', this._onAnalysisStatusChanged, this);
-      }
-    }, this);
+  _restartAnalysisPolling: function () {
+    this._analysisPoller.resetAnalysisNodes(this._getAnalysisNodeModels());
   },
 
-  _onAnalysisStatusChanged: function (analysisModel) {
-    if (analysisModel.isDone()) {
-      this.untrackLoadingObject(analysisModel);
-      if (this._isAnalysisSourceOfLayerOrDataview(analysisModel)) {
-        this.reload();
-      }
-    }
+  _getAnalysisNodeModels: function () {
+    return this._analysisCollection.models;
```

Comment:
@IagoLast instead of using `this._analysisCollection` here, we need to extract sources from `this._layersCollection` and `this._dataviewsCollection`.